### ISSUE TITLE
Tendrl python componetns fails to start when etcd tls authentication is used and etcd instance is referenced via ip address

### ISSUE THREAD

**Issue by mbukatov** [Tendrl/commons#759]
## Description

Tendrl python clients seem to ignore *X509v3 Subject Alternative Name extension* when client server tls authentication is used.

When etcd is configured to use server client tls based authentication (so that there is client tls certificate and private key on each machine where we have tendrl components which requires to access etcd) and etcd instance is referenced via ip address, tendrl component fails to authenticate with etcd, even though the tls certificate specifies the ip address via X509v3 Subject Alternative Name extension. This failure prevents affected tendrl service to start, which prevents Tendrl to provide any services.

The issue is created for tendrl-commons, as the same issue can be seen for multiple python tendrl components, such as tendrl-node-agent and tendrl-monitoring-integration.

When fqdn is used instead, the authentication works (see section Additional Evidence below).

Tendrl api service seems to be not affected, it is up and running even when etcd is referenced via ip address.

## Version

```
tendrl-commons-1.5.3-20171013T081843.c73101a.noarch
```

All related packages on Tendrl server machine:

```
tendrl-api-1.5.3-20171013T082716.a2f3b3f.noarch
tendrl-api-httpd-1.5.3-20171013T082716.a2f3b3f.noarch
tendrl-commons-1.5.3-20171013T081843.c73101a.noarch
tendrl-grafana-plugins-1.5.3-20171013T082252.ce16484.noarch
tendrl-monitoring-integration-1.5.3-20171013T082252.ce16484.noarch
tendrl-node-agent-1.5.3-20171013T081912.cdf0d6e.noarch
tendrl-notifier-1.5.3-20171011T200310.3c01717.noarch
tendrl-ui-1.5.3-20171013T082611.6e08356.noarch
python-etcd-0.4.5-1.noarch
etcd-3.2.5-1.el7.x86_64
rubygem-etcd-0.3.0-1.el7.centos.noarch
```

```
# cat /etc/redhat-release 
CentOS Linux release 7.4.1708 (Core)
```

## Steps to Reproduce

1. Prepare machines with GlusterFS cluster, including gluster volume (I used [nightly builds](https://github.com/usmqe/usmqe-setup/tree/master/roles/gluster-centos-repo) and [volume_alpha_distrep_4x2.create.conf](https://github.com/usmqe/usmqe-setup/blob/master/gdeploy_config/volume_alpha_distrep_4x2.create.conf))
2. Create and deploy tls certificate for all tendrl machines (server and all storage nodes), so that the client certificate has CN with fqdn, and ip address is referenced in X509v3 Subject Alternative Name extension (see https://github.com/usmqe/usmqe-setup/tree/master/roles/qe-ssl-cert).
3. Install Tendrl via tendrl-ansible there, using [snapshot builds](https://copr.fedorainfracloud.org/coprs/tendrl/tendrl/) and https://github.com/Tendrl/tendrl-ansible/pull/54/commits/4e443d883572dc1dd98cab7f8eb8625d24d8c6cc  from work in progress branch for https://github.com/Tendrl/tendrl-ansible/pull/54

## Actual Results

The installation via ansible fails on unsuccessful restart of some tendrl service, eg. tendrl-notifier:

```
RUNNING HANDLER [tendrl-server : restart tendrl-notifier] ****************************************************************************************************
Friday 13 October 2017  12:56:34 +0200 (0:00:00.893)       0:06:39.141 ********
fatal: [mbukatov-usm1-server.example.com]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to start service tendrl-notifier: Job for tendrl-notifier.service failed because start of the service was attempted too often. See \"systemctl status tendrl-notifier.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed tendrl-notifier.service\" followed by \"systemctl start tendrl-notifier.service\" again.\n"}  
        to retry, use: --limit @/home/martin/.ansible/retry-files/site.retry   
```

Looking into the journal reveals the problem as tls authentication failure (here we see a different component, to make it clear that this affects all python tendrl componetnts in the same way):

```
Oct 13 10:56:33 mbukatov-usm1-server.example.com systemd[1]: tendrl-node-agent.service holdoff time over, scheduling restart.
Oct 13 10:56:33 mbukatov-usm1-server.example.com systemd[1]: Started A python agent local to every managed storage node in the sds cluster.
Oct 13 10:56:33 mbukatov-usm1-server.example.com systemd[1]: Starting A python agent local to every managed storage node in the sds cluster...
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: Creating namespace.node_agent from source tendrl.node_agentnamespace.node_agent created!Finding objects in namespace.node_agent.objectsRegistering object namespace.node_agent.objects.ClusterAlertCountersFinding atoms in namespace.node_agent.objects.ClusterAlertCounters.atomsFinding flows in namespace.node_agent.objects.ClusterAlertCounters.flowsRegistering object namespace.node_agent.objects.ClusterMessageFinding atoms in namespace.node_agent.objects.ClusterMessage.atomsFinding flows in namespace.node_agent.objects.ClusterMessage.flowsRegistering object namespace.node_agent.objects.CompiledDefinitionsFinding atoms in namespace.node_agent.objects.CompiledDefinitions.atomsFinding flows in namespace.node_agent.objects.CompiledDefinitions.flowsRegistering object namespace.node_agent.objects.ConfigFinding atoms in namespace.node_agent.objects.Config.atomsFinding flows in namespace.node_agent.objects.Config.flowsRegistering object namespace.node_agent.objects.DefinitionFinding atoms in namespace.node_agent.objects.Definition.atomsFinding flows in namespace.node_agent.objects.Definition.flowsRegistering object namespace.node_agent.objects.GlobalNetworkFinding atoms in namespace.node_agent.objects.GlobalNetwork.atomsFinding flows in namespace.node_agent.objects.GlobalNetwork.flowsRegistering object namespace.node_agent.objects.MessageFinding atoms in namespace.node_agent.objects.Message.atomsFinding flows in namespace.node_agent.objects.Message.flowsRegistering object namespace.node_agent.objects.NodeAlertCountersFinding atoms in namespace.node_agent.objects.NodeAlertCounters.atomsFinding flows in namespace.node_agent.objects.NodeAlertCounters.flowsRegistering object namespace.node_agent.objects.NodeMessageFinding atoms in namespace.node_agent.objects.NodeMessage.atomsFinding flows in namespace.node_agent.objects.NodeMessage.flowsRegistering object namespace.node_agent.objects.NotificationMessageFinding atoms in namespace.node_agent.objects.NotificationMessage.atomsFinding flows in namespace.node_agent.objects.Notificati
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: onMessage.flowsFinding flows in namespace.node_agent.flowsRegistering flow namespace.node_agent.flows.GenerateJournalMappingSetup Tendrl definitions (.yml)for namespace.node_agentValidating registered (.py) flows in namespace.node_agent.flowsValidating defined (.yml) flows in namespace.node_agent.flowsValidating registered (.py) objects in namespace.node_agent.objectsValidating defined (.yml) objects in namespace.node_agent.objectsSetup Config for namespace.node_agentSetup central store clients for namespace.node_agentCreating namespace.tendrl from source tendrl.commonsnamespace.tendrl created!Finding objects in namespace.tendrl.objectsRegistering object namespace.tendrl.objects.AlertFinding atoms in namespace.tendrl.objects.Alert.atomsFinding flows in namespace.tendrl.objects.Alert.flowsRegistering object namespace.tendrl.objects.BlockDeviceFinding atoms in namespace.tendrl.objects.BlockDevice.atomsFinding flows in namespace.tendrl.objects.BlockDevice.flowsRegistering object namespace.tendrl.objects.ClusterFinding atoms in namespace.tendrl.objects.Cluster.atomsRegistering atom namespace.tendrl.objects.Cluster.atoms.CheckClusterAvailableRegistering atom namespace.tendrl.objects.Cluster.atoms.ConfigureMonitoringRegistering atom namespace.tendrl.objects.Cluster.atoms.CreateRegistering atom namespace.tendrl.objects.Cluster.atoms.ImportClusterRegistering atom namespace.tendrl.objects.Cluster.atoms.ImportCreatedClusterRegistering atom namespace.tendrl.objects.Cluster.atoms.PersistClusterDetailsAfterCreateRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidCreateClusterParamsRegistering atom namespace.tendrl.objects.Cluster.atoms.ValidImportClusterParamsFinding flows in namespace.tendrl.objects.Cluster.flowsRegistering object namespace.tendrl.objects.AlertFinding atoms in namespace.tendrl.objects.Alert.atomsFinding flows in namespace.tendrl.objects.Alert.flowsRegistering object namespace.tendrl.objects.ClusterAlertFinding atoms in namespace.tendrl.objects.ClusterAlert.atomsFinding flows in namespace.tendrl.obj
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: ects.ClusterAlert.flowsRegistering object namespace.tendrl.objects.ClusterNodeContextFinding atoms in namespace.tendrl.objects.ClusterNodeContext.atomsFinding flows in namespace.tendrl.objects.ClusterNodeContext.flowsRegistering object namespace.tendrl.objects.ClusterTendrlContextFinding atoms in namespace.tendrl.objects.ClusterTendrlContext.atomsFinding flows in namespace.tendrl.objects.ClusterTendrlContext.flowsRegistering object namespace.tendrl.objects.CpuFinding atoms in namespace.tendrl.objects.Cpu.atomsFinding flows in namespace.tendrl.objects.Cpu.flowsRegistering object namespace.tendrl.objects.DefinitionFinding atoms in namespace.tendrl.objects.Definition.atomsFinding flows in namespace.tendrl.objects.Definition.flowsRegistering object namespace.tendrl.objects.DetectedClusterFinding atoms in namespace.tendrl.objects.DetectedCluster.atomsFinding flows in namespace.tendrl.objects.DetectedCluster.flowsRegistering object namespace.tendrl.objects.DiskFinding atoms in namespace.tendrl.objects.Disk.atomsFinding flows in namespace.tendrl.objects.Disk.flowsRegistering object namespace.tendrl.objects.JobFinding atoms in namespace.tendrl.objects.Job.atomsFinding flows in namespace.tendrl.objects.Job.flowsRegistering object namespace.tendrl.objects.MemoryFinding atoms in namespace.tendrl.objects.Memory.atomsFinding flows in namespace.tendrl.objects.Memory.flowsRegistering object namespace.tendrl.objects.NodeFinding atoms in namespace.tendrl.objects.Node.atomsRegistering atom namespace.tendrl.objects.Node.atoms.BaseAtomRegistering atom namespace.tendrl.objects.Node.atoms.CheckNodeUpRegistering atom namespace.tendrl.objects.Node.atoms.BaseAtomRegistering atom namespace.tendrl.objects.Node.atoms.CmdRegistering atom namespace.tendrl.objects.Node.atoms.IsNodeTendrlManagedFinding flows in namespace.tendrl.objects.Node.flowsRegistering object namespace.tendrl.objects.AlertFinding atoms in namespace.tendrl.objects.Alert.atomsFinding flows in namespace.tendrl.objects.Alert.flowsRegistering object namespace.tendrl.objects.Nod
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: eAlertFinding atoms in namespace.tendrl.objects.NodeAlert.atomsFinding flows in namespace.tendrl.objects.NodeAlert.flowsRegistering object namespace.tendrl.objects.NodeContextFinding atoms in namespace.tendrl.objects.NodeContext.atomsFinding flows in namespace.tendrl.objects.NodeContext.flowsRegistering object namespace.tendrl.objects.NodeNetworkFinding atoms in namespace.tendrl.objects.NodeNetwork.atomsFinding flows in namespace.tendrl.objects.NodeNetwork.flowsRegistering object namespace.tendrl.objects.AlertFinding atoms in namespace.tendrl.objects.Alert.atomsFinding flows in namespace.tendrl.objects.Alert.flowsRegistering object namespace.tendrl.objects.NotificationOnlyAlertFinding atoms in namespace.tendrl.objects.NotificationOnlyAlert.atomsFinding flows in namespace.tendrl.objects.NotificationOnlyAlert.flowsRegistering object namespace.tendrl.objects.OsFinding atoms in namespace.tendrl.objects.Os.atomsFinding flows in namespace.tendrl.objects.Os.flowsRegistering object namespace.tendrl.objects.PlatformFinding atoms in namespace.tendrl.objects.Platform.atomsFinding flows in namespace.tendrl.objects.Platform.flowsRegistering object namespace.tendrl.objects.ServiceFinding atoms in namespace.tendrl.objects.Service.atomsRegistering atom namespace.tendrl.objects.Service.atoms.BaseAtomRegistering atom namespace.tendrl.objects.Service.atoms.CheckServiceStatusFinding flows in namespace.tendrl.objects.Service.flowsRegistering object namespace.tendrl.objects.TendrlContextFinding atoms in namespace.tendrl.objects.TendrlContext.atomsFinding flows in namespace.tendrl.objects.TendrlContext.flowsRegistering object namespace.tendrl.objects.DiskFinding atoms in namespace.tendrl.objects.Disk.atomsFinding flows in namespace.tendrl.objects.Disk.flowsRegistering object namespace.tendrl.objects.VirtualDiskFinding atoms in namespace.tendrl.objects.VirtualDisk.atomsFinding flows in namespace.tendrl.objects.VirtualDisk.flowsFinding flows in namespace.tendrl.flowsRegistering flow namespace.tendrl.flows.AuthorizeSshKeyRegistering flow
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-age[11218]: Starting new HTTPS connection (1): 10.37.169.90
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: namespace.tendrl.flows.ConfigureMonitoringRegistering flow namespace.tendrl.flows.CreateClusterRegistering flow namespace.tendrl.flows.ExpandClusterRegistering flow namespace.tendrl.flows.ImportClusterRegistering flow namespace.tendrl.flows.SetupSshSetup Tendrl definitions (.yml)for namespace.tendrlValidating registered (.py) flows in namespace.tendrl.flowsValidating defined (.yml) flows in namespace.tendrl.flowsValidating registered (.py) objects in namespace.tendrl.objectsValidating defined (.yml) objects in namespace.tendrl.objectsSetup NodeContext for namespace.tendrlLoad definitions (.yml) for namespace.tendrl.objects.NodeContextStarting new HTTPS connection (1): 10.37.169.90
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-age[11218]: Request to server https://10.37.169.90:2379 failed: SSLError(CertificateError("hostname '10.37.169.90' doesn't match 'mbukatov-usm1-server.example.com'",),)
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: Request to server https://10.37.169.90:2379 failed: SSLError(CertificateError("hostname '10.37.169.90' doesn't match 'mbukatov-usm1-server.example.com'",),)
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: Traceback (most recent call last):
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: File "/usr/bin/tendrl-node-agent", line 9, in <module>
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: load_entry_point('tendrl-node-agent==1.5.3', 'console_scripts', 'tendrl-node-agent')()
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: File "/usr/lib/python2.7/site-packages/tendrl/node_agent/manager/__init__.py", line 41, in main
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: TendrlNS()
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: File "/usr/lib/python2.7/site-packages/tendrl/commons/__init__.py", line 45, in __init__
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: self.setup_common_objects()
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: File "/usr/lib/python2.7/site-packages/tendrl/commons/__init__.py", line 300, in setup_common_objects
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: self.current_ns.objects.NodeContext()
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: File "/usr/lib/python2.7/site-packages/tendrl/commons/objects/node_context/__init__.py", line 34, in __init__
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: self.node_id).value
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: File "/usr/lib/python2.7/site-packages/etcd/client.py", line 597, in read
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: timeout=timeout)
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: File "/usr/lib/python2.7/site-packages/etcd/client.py", line 893, in wrapper
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: cause=e
Oct 13 10:56:34 mbukatov-usm1-server.example.com tendrl-node-agent[11218]: etcd.EtcdConnectionFailed: Connection to etcd failed due to SSLError(CertificateError("hostname '10.37.169.90' doesn't match 'mbukatov-usm1-server.example.com'",),)
Oct 13 10:56:34 mbukatov-usm1-server.example.com systemd[1]: tendrl-node-agent.service: main process exited, code=exited, status=1/FAILURE
Oct 13 10:56:34 mbukatov-usm1-server.example.com systemd[1]: Unit tendrl-node-agent.service entered failed state.
Oct 13 10:56:34 mbukatov-usm1-server.example.com systemd[1]: tendrl-node-agent.service failed.
```

## Expected Results

Tendrl components are able to authenticate with etcd using ip addres when it's included in X509v3 Subject Alternative Name.

## Additional information

In our lab, we have machines with proper dns entries configured for all machines and our own certification authority, which we use to issue client certificates. We try to follow best practices and mimic the real world setup as much as possible.

The actuall error (extracted from the journal output above) looks like this:

```
etcd.EtcdConnectionFailed: Connection to etcd failed due to SSLError(CertificateError("hostname '10.37.169.90' doesn't match 'mbukatov-usm1-server.example.com'",),)
```

Note that the client certificate deployed on all machines looks like this:

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 11057 (0x2b31)
    Signature Algorithm: sha1WithRSAEncryption
        Issuer: CN=ooo.englab.brq.redhat.com
        Validity
            Not Before: Oct 13 10:34:19 2017 GMT
            Not After : Oct 13 10:34:19 2019 GMT
        Subject: C=CZ, ST=CZ, O=Red Hat, OU=USM - QE, CN=mbukatov-usm1-server.example.com
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:b5:ec:3e:5c:5f:9d:f2:d4:95:10:90:c5:22:2f:
                    c9:eb:b2:b3:ad:f0:30:17:07:a7:17:aa:0e:49:fa:
                    99:ef:e5:da:15:8b:de:f5:a7:37:28:a1:75:fc:53:
                    0f:10:e3:20:67:12:7b:8a:3d:6b:95:83:a6:b7:a9:
                    92:c1:db:bd:2c:ac:37:77:3d:36:4f:71:80:ff:15:
                    18:cb:27:51:32:f6:d6:c3:4f:33:00:92:3d:1d:75:
                    f7:69:84:0c:4f:18:e8:24:e0:12:a3:d0:7d:7a:b2:
                    25:ef:5a:01:ab:d2:5f:64:18:e0:37:7c:14:cc:f5:
                    dc:91:31:94:c4:35:98:60:42:f5:be:e2:33:b7:d5:
                    5f:a8:d2:70:49:1d:90:eb:6b:38:29:52:ff:ab:bc:
                    6a:79:1f:d3:76:25:76:ee:a1:a9:18:f6:ae:fa:5c:
                    04:3d:ab:59:3d:70:27:e9:a2:b2:d9:56:2f:1f:1f:
                    ac:32:5b:45:7b:aa:2e:1d:dc:80:c0:ca:f6:0e:36:
                    b5:db:f0:0f:76:94:ec:4a:32:1e:4a:3c:dd:e3:f7:
                    4d:f4:7c:7c:4a:d6:d7:ea:a1:d2:94:7f:43:fc:c7:
                    5a:2b:bd:25:75:ad:66:32:b7:1e:b1:99:85:80:53:
                    4d:03:2c:5d:86:72:14:77:37:76:e7:95:58:ae:ce:
                    7f:d7
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Basic Constraints:
                CA:FALSE
            Netscape Comment:
                OpenSSL Generated Certificate
            X509v3 Subject Key Identifier:
                BD:F0:CB:10:16:24:8B:BE:1E:B4:2A:A2:F3:10:EE:B9:A3:FB:77:6E
            X509v3 Authority Key Identifier:
                DirName:/CN=ooo.englab.brq.redhat.com
                serial:A2:3B:A0:37:90:58:1A:14

            X509v3 Key Usage:
                Digital Signature, Non Repudiation, Key Encipherment
            X509v3 Subject Alternative Name:
                DNS:mbukatov-usm1-server.example.com, IP Address:10.37.169.90
    Signature Algorithm: sha1WithRSAEncryption
         3f:31:b1:e5:3d:a3:bb:42:e8:22:e7:c3:56:b6:a6:43:56:a3:
         d7:eb:d1:53:21:5e:fb:03:7d:97:4c:79:f7:a6:8b:74:8c:bc:
         61:32:91:90:1e:b3:61:8c:68:77:32:03:19:8d:9f:94:29:ae:
         4f:b2:b2:f3:13:85:49:ab:d6:62:bc:b8:bd:09:53:ee:81:4e:
         41:91:a7:c8:b7:af:11:3e:61:31:30:13:a8:96:bb:22:e7:fe:
         55:7d:05:86:3d:f5:97:6e:28:73:9a:93:1f:c6:8b:2e:08:98:
         9f:a2:a0:b6:84:9b:28:f8:92:77:3f:fe:50:71:17:27:07:09:
         4f:51:41:f6:b4:0e:83:35:c2:4e:04:48:06:8f:f6:c1:12:99:
         aa:92:f0:9e:a0:70:d1:c9:4e:2c:39:35:be:9b:a0:bd:8c:d7:
         79:0c:b4:69:7c:8d:97:93:91:08:f3:02:5b:af:40:be:2b:d8:
         5c:30:f8:1b:7f:c5:5a:6b:f9:9c:be:c4:04:0f:f1:da:01:09:
         03:4f:e3:3d:b4:74:b7:c8:01:4c:5d:db:45:36:70:08:e7:3f:
         45:4c:12:28:2c:9d:9b:67:13:c3:22:fa:b7:89:da:9d:99:d8:
         27:1d:a1:6d:89:a2:0e:81:78:f8:6f:5c:07:97:0e:31:e2:e2:
         f0:d9:8f:c7

```

Note that IP address is referenced via X509v3 Subject Alternative Name.

Also note that I'm able to connect to etcd via both curl and etcdctl, using both fqdn and ip address:

```
# curl --cacert /etc/pki/tls/certs/ca-usmqe.crt --cert /etc/pki/tls/certs/etcd.crt --key /etc/pki/tls/private/etcd.key https://10.37.169.90:2379/v2/keys/
# curl --cacert /etc/pki/tls/certs/ca-usmqe.crt --cert /etc/pki/tls/certs/etcd.crt --key /etc/pki/tls/private/etcd.key https://mbukatov-usm1-server.example.com:2379/v2/keys/
# etcdctl --ca-file /etc/pki/tls/certs/ca-usmqe.crt --cert-file /etc/pki/tls/certs/etcd.crt --key-file /etc/pki/tls/private/etcd.key --endpoints https://mbukatov-usm1-server.example.com:2379 ls /
# etcdctl --ca-file /etc/pki/tls/certs/ca-usmqe.crt --cert-file /etc/pki/tls/certs/etcd.crt --key-file /etc/pki/tls/private/etcd.key --endpoints https://10.37.169.90:2379 ls /
```

In other words, both curl and etcdctl is able to authenticate using client tls certificate in both cases without any problems, which is a hint that the tls setup has been done correctly.

## Additional Evidence

When I reconfigure eg. node agent to use fqdn instead of ip address for etcd:

```
# grep etcd_connectio /etc/tendrl/node-agent/node-agent.conf.yaml
# etcd_connection: 10.37.169.90
etcd_connection: mbukatov-usm1-server.example.com
# systemctl start tendrl-node-agent
# journalctl -f -u tendrl-node-agent
```

and start the service again, it authenticates correctly and doesn't immediately crash.

**Comment by nthomas-redhat** [comment]
@mbukatov , is this issue still valid?

**Comment by r0h4n** [comment]
Closing since there is not verification
